src/Microsoft.Azure.WebJobs.Host/Queues/QueueProcessor.cs

```diff
@@ -142,6 +136,30 @@ public QueueProcessor(QueueProcessorFactoryContext context)
             }
         }
 
+        /// <summary>
+        /// Copies properties from the specified message and moves it to the poison queue.
+        /// </summary>
+        /// <param name="message">The poison message</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use</param>
+        /// <returns></returns>
+        internal async Task MoveMessageToPoisonQueueAsync(CloudQueueMessage message, CancellationToken cancellationToken)
```

Comment:
nit: maybe rename this to HandlePoisonMessageAsync, to avoid the similar name to CopyToPoisonQueueAsync. Handle makes sense, since we're doing multiple operations - copying the message to poison queue then deleting it.